### PR TITLE
Add Float 16 Support

### LINKED ISSUE #485: support 16bit float, a.k.a. half type
The tricky part of this is that C/C++ don't have native support for it.
There are various headers-only libraries for maniuplating them, including one in SPIRV-Tools that awoloszyn wrote

### PR DESCRIPTION
**Pull request by dj2**
This CL adds the `float16` data type into amber_script and the required plumbing to enable the feature.

Fixes #485

### CODE REVIEW

File: samples/config_helper_vulkan.cc [hunk 1]
```diff
@@ -46,6 +46,7 @@ const size_t kNumberOfRequiredValidationLayers =
 const char kVariablePointers[] = "VariablePointerFeatures.variablePointers";
 const char kVariablePointersStorageBuffer[] =
     "VariablePointerFeatures.variablePointersStorageBuffer";
+const char kFloat16Int8[] = "Float16Int8Features.shaderFloat16";
```
Comment (alan-baker) [inline samples/config_helper_vulkan.cc]: This should probably be kFloat16Int8Float16 or something equally ridiculous if you want to eventually support the int8 feature bit.

File: src/float16_helper.cc [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/float16_helper.h"
+
+#include <cassert>
+#include <iostream>
```
Comment (alan-baker) [inline src/float16_helper.cc]: Left over debugging?

File: samples/config_helper_vulkan.cc (hunk 1 above)
Comment (dj2) [reply samples/config_helper_vulkan.cc]: Done.

File: src/float16_helper.cc (hunk 2 above)
Comment (dj2) [reply src/float16_helper.cc]: done

**Pull request merged**